docs/en/03_Upgrading/index.md

```diff
@@ -10,9 +10,66 @@ your SilverStripe installation means overwriting files, flushing the cache and u
 See our [upgrade notes and changelogs](/changelogs/4.0.0) for 4.0.0 specific information, bugfixes and API changes.
 </div>
 
-## Composer
+## Understanding and planning your upgrade {#planning}
+
+How easy will it be to update my project? It's a fair question, and sometimes a difficult one to answer.
+
+* SilverStripe follows _semantic versioning_ (see our [release process](/contributing/release_process) for details).
+  * "Major" releases introduces API change that may break your application.
+  * "Minor" releases (x.y) introduces API changes in a backward compatible way and can mark some API as deprecated.
+  * "Patch" releases (x.y.z) fix bugs without introducing any API changes.
+* If you've made custom branches of SilverStripe core, or any thirdparty module, it's going to be harder to upgrade.
+* The more custom features you have, the harder it will be to upgrade. You will have to re-test all of those features, and adapt to API changes in core.
+* Customizations of a well defined type - such as custom page types or custom blog widgets - are going to be easier to upgrade than customisations that modify deep system internals like rewriting SQL queries.
+
+### Overview of changes
+
+If you've already had a look over the changelog, you will see that there are some fundamental changes that need to be implemented to upgrade from 3.x. Here's a couple of the most important ones to consider:
+
+* PHP 5.6 is now the minimum required version (and up to PHP 7.2.x is supported!).
+* CMS CSS has been re-developed using Bootstrap 4 as a base.
+* SilverStripe code _should_ now be [PSR-2 compliant](http://www.php-fig.org/psr/psr-2/). While this is not a requirement, we strongly suggest you switch over now. You can use tools such as [`phpcbf`](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Fixing-Errors-Automatically) to do most of it automatically.
+* We've introduced some best practices for module development. [See the Modules article](/developer_guides/extending/modules) for more information.
+
+[Read the SilverStripe 4.0.0 changelogs](/changelogs/4.0.0/) for a comprehensive list of changes introduced with SilverStripe 4.
```

Comment:
Should link to 4.1.0 and 4.2.0 as well

Reply:
I've linked to the list of important API changes at the end.